Sources/SwiftBSON/ExtendedJSONDecoder.swift

```diff
@@ -28,21 +56,138 @@ public class ExtendedJSONDecoder {
     /// - Returns: Decoded representation of the JSON input as an instance of `T`.
     /// - Throws: `DecodingError` if the JSON data is corrupt or if any value throws an error during decoding.
     public func decode<T: Decodable>(_: T.Type, from data: Data) throws -> T {
-        // Data --> JSON --> BSON --> T
+        // Data --> JSONValue --> BSON --> T
         // Takes in JSON as `Data` encoded with `.utf8` and runs it through a `JSONDecoder` to get an
         // instance of the `JSON` enum.
-
-        // In earlier versions of Swift, JSONDecoder doesn't support decoding "fragments" at the top level, so we wrap
-        // the data in an array to guarantee it always decodes properly.
-        let wrappedData = "[".utf8 + data + "]".utf8
-        let json = try JSONDecoder().decode([JSON].self, from: wrappedData)[0]
+        let json = try JSONParser().parse(bytes: data)
 
         // Then a `BSON` enum instance is created via the `JSON`.
-        let bson = try BSON(fromExtJSON: json, keyPath: [])
+        // let bson = try json.toBSON(keyPath: [])
+        let bson = try self.decodeBSONFromJSON(json, keyPath: [])
```

Comment:
I just realized that I forgot to include the error conversions to `DecodingError` in here, will submit a patch shortly to fix this.